starcheck/calc_ccd_temps.py

```diff
@@ -204,23 +204,25 @@ def get_interval_data(intervals, times, ccd_temp, obsreqs=None):
         start_idx = -1 + np.searchsorted(times, interval['tstart'])
         ok_temps = ccd_temp[start_idx:stop_idx]
         ok_times = times[start_idx:stop_idx]
-        if len(ok_temps) > 0:
-            obs['ccd_temp'] = np.max(ok_temps)
-            obs['n100_warm_frac'] = dark_model.get_warm_fracs(
-                100, interval['tstart'], np.max(ok_temps))
-        if obsreqs is None:
-            obsreqs = {}
-        if interval['obsid'] in obsreqs and len(ok_temps) > 0:
+        # If there are no good samples, put the times in the output dict and go to the next interval
+        if len(ok_temps) == 0:
+            obstemps[str(interval['obsid'])] = obs
+            continue
+        obs['ccd_temp'] = np.max(ok_temps)
+        obs['n100_warm_frac'] = dark_model.get_warm_fracs(
+            100, interval['tstart'], np.max(ok_temps))
+        # If we have an OR list, the obsid is in that list, and the OR list has zero-offset keys
+        if (obsreqs is not None and interval['obsid'] in obsreqs
+                and 'chip_id' in obsreqs[interval['obsid']]):
```

Comment:
I deviated slightly from your recommendations here @taldcroft and tried to simplify the nested conditionals in this routine.